raiden/raiden_service.py

```diff
@@ -476,48 +478,87 @@ def handle_state_change(self, state_change: StateChange):
             state_change=_redact_secret(serialize.JSONSerializer.serialize(state_change)),
         )
 
-        event_list = self.wal.log_and_dispatch(state_change)
+        raiden_event_list = self.wal.log_and_dispatch(state_change)
 
-        if self.dispatch_events_lock.locked():
-            return []
+        log.debug(
+            'Raiden events',
+            node=pex(self.address),
+            raiden_events=[
+                _redact_secret(serialize.JSONSerializer.serialize(event))
+                for event in raiden_event_list
+            ],
+        )
 
-        for event in event_list:
-            log.debug(
-                'Raiden event',
-                node=pex(self.address),
-                raiden_event=_redact_secret(serialize.JSONSerializer.serialize(event)),
+        if not self.dispatch_events_lock.locked():
```

Comment:
If the lock is already held, isn't this skipping `if` skipping the returned `raiden_event_list` completely? I can't see it queued elsewhere

Reply:
yes, that is what the lock is for:

https://github.com/raiden-network/raiden/blob/a9bb006db4ce3e9ae672a1379c2a11f0f49cc3c2/raiden/raiden_service.py#L252-L255

Reply:
Yes, but "prevent from processing" and "discarding returned events completely" seems very different to me. Are these events queued somewhere else? Or are they safe to be discarded this way?

Reply:
Yes, the events that have external side effects are kept in the `ChainState` object in the queues `pending_transactions` and `queueids_to_queues:

https://github.com/raiden-network/raiden/blob/00a93341dfc2443a605f71e8a2c43d329651ddaf/raiden/transfer/state.py#L276-L278

Ignoring the events is required here, more info:

https://github.com/raiden-network/raiden/pull/3230